### PR TITLE
[risk=no] Fix a UI bug: could not edit WS name in the middle of the field

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

Attempting to edit a workspace name does not work correctly.  Here I'm succeeding at adding "end is fine" but failing at updating in the middle with "middle is not"

middle edit bad

This is similar behavior to something we've seen before with Duplicate Workspace, and the fix is the same.  Not sure why we didn't fix both then.

Editing in the middle now works:

middle_now_ok

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI + API server with `yarn test-local`
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 1]
```diff
@@ -614,15 +614,16 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
...
-      switch (this.props.routeConfigData.mode) {
+      // use workspace name from props instead of state here
+      // because it's a record of the initial value
+      const {routeConfigData: {mode}, workspace: {name}} = this.props;
+      switch (mode) {
         case WorkspaceEditMode.Create:
           return 'Create a new workspace';
         case WorkspaceEditMode.Edit:
-          return 'Edit workspace \"' + this.state.workspace.name + '\"';
+          return 'Edit workspace \"' + name + '\"';
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:624]: Now always shows "Edit workspace `oldname`" instead of "Edit workspace `constantly updating newname`"

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 2]
```diff
@@ -1026,7 +1027,7 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
                                 required={!this.isMode(WorkspaceEditMode.Duplicate)}>
           <FlexRow style={{alignItems: 'baseline'}}>
             <TextInput type='text' style={styles.textInput} autoFocus placeholder='Workspace Name'
-              value = {name}
+              defaultValue = {name}
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:1030]: not 100% sure why this change works, but it seems to.  Please LMK if I have misunderstood something here.

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 3]
```diff
@@ -614,15 +614,16 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
...
-      switch (this.props.routeConfigData.mode) {
+      // use workspace name from props instead of state here
+      // because it's a record of the initial value
+      const {routeConfigData: {mode}, workspace} = this.props;
+      switch (mode) {
         case WorkspaceEditMode.Create:
           return 'Create a new workspace';
         case WorkspaceEditMode.Edit:
-          return 'Edit workspace \"' + this.state.workspace.name + '\"';
+          return 'Edit workspace \"' + workspace.name + '\"';
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:624]: Now always shows "Edit workspace `oldname`" instead of the continuously changing "Edit workspace `newname`"

**Review by petesantos — APPROVED** [review]
LGTM - see comment.

File: ui/src/app/pages/workspace/workspace-edit.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmthibault79) [reply ui/src/app/pages/workspace/workspace-edit.tsx]: Thanks!  This makes sense but I would not have guessed it works this way.  It does continue to work with `value`.

**Review by petesantos — APPROVED** [review]
LGTM
